File: NC Classic/LC2/Solution.cpp
class Solution {
   public:
    /**
     *
     * @param tokens string字符串vector
     * @return int整型
     */
    int evalRPN(vector<string>& tokens) {
        stack<int> res;  //创建栈
        //遍历每一个输入，注意是字符串类型
        for (auto x : tokens) {
            if (x == "+" || x == "-" || x == "*" || x == "/") {
                //如果当前字符串是操作符
                if (res.size() < 2)
                    return 0;  //如果栈内数字不足两个，即无法进行运算，则返回0
                //依次将栈内两个数字出栈进行操作运算
                int a = res.top();
                res.pop();
                int b = res.top();
                res.pop();
                int c = 0;  //用c表示每次运算的结果
                if (x == "+")
                    c = b + a;
                else if (x == "-")
                    c = b - a;
                else if (x == "*")
                    c = b * a;
                else if (x == "/")
                    c = b / a;

                res.push(c);
            } else {
                //如果当前字符串是数字
                res.push(atoi(x.c_str()));
            }
        }
        return res.top();
    }
};

File: NC Classic/LC2/Solution.js
/**
  * 
  * @param tokens string字符串一维数组 
  * @return int整型
  */
function evalRPN(tokens) {
    let flag = ['+', '-', '*', '/'];
    var res = [];
    for (const ch of tokens) {
        let index = flag.indexOf(ch);
        if (index !== -1) {
            const b = res.pop();
            const a = res.pop();
            const op = flag[index];
            const sum = eval(`${a}${op}(${b})`);
            res.push(parseInt(sum));    // 除法运算，sum可能出现浮点数，根据题目要求，统一转为整数
        } else {
            res.push(ch);
        }
    }

    return parseInt(res.pop());
}
module.exports = {
    evalRPN: evalRPN
};